Cabal-syntax/src/Distribution/Types/VersionInterval.hs

```diff
@@ -70,6 +71,12 @@ newtype VersionIntervals = VersionIntervals [VersionInterval]
 unVersionIntervals :: VersionIntervals -> [VersionInterval]
 unVersionIntervals (VersionIntervals is) = is
 
+-- | Directly construct a 'VersionIntervals' from a list of intervals.
+mkVersionIntervals :: [VersionInterval] -> Maybe VersionIntervals
```

Comment:
FWIW, this is different signature than in the legacy module. That one doesn't have `Maybe`.

Also if the goal is to make `[VersionInterval] -> VersionRange` function, why just not export that, i.e. a variant of `fromVersionIntervals` with that signature. It works even if invariants are violated (e.g. if intervals overlap).